testing/internal/e2e/scope/scope_test.go

```diff
@@ -0,0 +1,322 @@
+package scope_test
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/boundary/api/authmethods"
+	"github.com/hashicorp/boundary/api/scopes"
+	"github.com/hashicorp/boundary/testing/internal/e2e"
+	"github.com/hashicorp/boundary/testing/internal/e2e/boundary"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestKeyDestructionCli uses the boundary CLI to test key destruction.
+func TestKeyDestructionCli(t *testing.T) {
+	e2e.MaybeSkipTest(t)
+
+	output := boundary.AuthenticateCli()
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	output = e2e.RunCommand(
+		"boundary", "scopes", "create",
+		"-name", "testscope",
+		"-format", "json",
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+
+	var scopeReply scopes.ScopeCreateResult
+	require.NoError(t, json.Unmarshal(output.Stdout, &scopeReply))
+	t.Cleanup(func() {
+		output = e2e.RunCommand(
+			"boundary", "scopes", "delete",
+			"-id", scopeReply.Item.Id,
+			"-format", "json",
+		)
+		require.NoError(t, output.Err, string(output.Stderr))
+	})
+
+	output = e2e.RunCommand(
+		"boundary", "scopes", "list-keys",
+		"-scope-id", scopeReply.Item.Id,
+		"-format", "json",
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+	var keys scopes.KeyListResult
+	err := json.Unmarshal(output.Stdout, &keys)
+	require.NoError(t, err)
+	assert.Len(t, keys.Items, 7)
+	for _, key := range keys.Items {
+		assert.Len(t, key.Versions, 1)
+	}
+
+	// Create OIDC auth method to create some encrypted data in the scope
+	output = e2e.RunCommand(
+		"boundary", "auth-methods", "create", "oidc",
+		"-scope-id", scopeReply.Item.Id,
+		"-api-url-prefix", "http://example.com",
+		"-client-id", "something",
+		"-client-secret", "somethingelse",
+		"-format", "json",
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+	var amReply authmethods.AuthMethodCreateResult
+	require.NoError(t, json.Unmarshal(output.Stdout, &amReply))
+	t.Cleanup(func() {
+		output = e2e.RunCommand(
+			"boundary", "auth-methods", "delete",
+			"-id", amReply.Item.Id,
+		)
```

Comment:
nit: If you wanted to, you could change this to one line since it's shorter. I'm not sure what the preferred styling is here.
```suggestion
		output = e2e.RunCommand("boundary", "auth-methods", "delete", "-id", amReply.Item.Id)
```

Reply:
Separate question, if we're deleting the scope/org in cleanup, do we need to explicitly clean up other things in that scope/org?

Reply:
I like to be explicit about it, even though it probably doesn't need to be done 😄